bip-PSBT-SP.mediawiki

```diff
@@ -0,0 +1,187 @@
+<pre>
+  BIP: ?
+  Layer: Applications
+  Title: Sending Silent Payments with PSBTs
+  Author: Andrew Toth <andrewstoth@gmail.com>
+          Ava Chow <me@achow101.com>
+          josibake <josibake@protonmail.com>
+  Comments-Summary: No comments yet.
+  Comments-URI: TBD
+  Status: Draft
+  Type: Standards Track
+  Created: 2024-05-14
+  License: BSD-2-Clause
+</pre>
+
+==Introduction==
+
+===Abstract===
+
+This document proposes additional fields and updated role responsibilities for BIP 370 PSBTv2
+which adds support for sending to silent payments as described in BIP352.
+
+===Copyright===
+
+This BIP is licensed under the 2-clause BSD license.
+
+===Motivation===
+
+Partially Signed Bitcoin Transaction Version 2 as described in BIP 370 is not compatible with sending to silent payments as described in BIP352. In particular, the output script of a silent payment cannot be computed until after all transaction inputs have been added.
+Also, any inputs that the Signer has the private keys for must be signed with SIGHASH_ALL and all inputs must not have any scriptPubKeys with Segwit version > 1.
+Additionally, the silent payment outputs computed by a signer must be verifiable to other entities.
+Therefore, new fields and role responsibilities must be added to carry, compute, and verify the silent payment data.
+
+==Specification==
+
+This document specifies new fields and new field inclusion/exclusion requirements.
+
+<tt>PSBT_OUT_SCRIPT</tt> is modified to be optional for outputs in silent payments capable PSBTs. If this field is not included in the output, then the field PSBT_OUT_SP_V0_INFO must be included.
```

Comment:
Does it make sense to require the presence of either `PSBT_OUT_SCRIPT` or `PSBT_OUT_SP_V0_INFO`? What if there is another future extension of BIP 370 that provides a third option?

Would it perhaps be sufficient to declare an output that has both the fields `PSBT_OUT_SCRIPT` and `PSBT_OUT_SP_V0_INFO` to make the PSBT invalid (i.e. at most one of these two can be present)?

Perhaps the interplay of these two fields could be elaborated.

Reply:
I don't think making these fields exclusive will work. The `PSBT_OUT_SCRIPT` must be added to sign the PSBT. The `PSBT_OUT_SP_V0_INFO` must be present to verify that the `PSBT_OUT_SCRIPT` is actually correct after it is added.

> What if there is another future extension of BIP 370 that provides a third option?

That would indeed soften this requirement. This BIP would have to be modified to support that extension, making it

> If either `PSBT_OUT_SCRIPT` or `PSBT_OUT_NEW_EXTENSION` are not included in the output, then the field PSBT_OUT_SP_V0_INFO must be included.

I don't really have an idea on how this could be made more graceful for that scenario.

> Perhaps the interplay of these two fields could be elaborated.

I will add some more context in a footnote, thanks.

Reply:
Clarified this section and added a footnote.